app/src/main/java/org/ieeeguc/ieeeguc/fragments/user/UserShow.java

```diff
@@ -1,24 +1,98 @@
 import android.view.ViewGroup;
-
+import android.widget.TextView;
+import org.ieeeguc.ieeeguc.HTTPResponse;
 import org.ieeeguc.ieeeguc.R;
+import org.ieeeguc.ieeeguc.controllers.MainActivity;
+import org.json.JSONObject;
+
 
 /**
  * A simple {@link Fragment} subclass.
  */
 public class UserShow extends Fragment {
 
+    int id=0;
+    String type="";
+    String first_name="";
+    String last_name="";
+    String email="";
+    String gender="";
+    String phone_number="";
+    String birthdate="";
+    int IEEE_membership_ID=0;
+    //JSONObject settings ;
+
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_user_show, container, false);
+
+
+        final View view = inflater.inflate(R.layout.fragment_user_show, container, false);
+        MainActivity.loggedInUser.getUser(MainActivity.token,MainActivity.loggedInUser.getId(), new HTTPResponse() {
+            @Override
+            public void onSuccess(int statusCode, JSONObject body) {
+                try {
+                    JSONObject j = body.getJSONObject("result");
+                    id = j.getInt("id");
+                    type = j.getString("type");
+                    first_name  = j.getString("first_name");
+                    last_name = j.getString("last_name");
+                    email = j.getString("email");
+                    gender = j.getString("gender");
+                    phone_number = j.getString("phone_number");
+                    birthdate = j.getString("birthdate");
+                    IEEE_membership_ID = j.getInt("IEEE_membership_ID");
+                    //settings = j.getJSONObject("setting");
+
+
+
+
+                } catch (Exception e) {
+
+                    MainActivity.createSnackBar(getString(R.string.error_server_down));
+                }
+
+                ((TextView) view.findViewById(R.id.name)).setText(first_name+" "+last_name);
+                ((TextView) view.findViewById(R.id.email)).setText(email);
+                ((TextView) view.findViewById(R.id.id)).setText(id+"");
+                ((TextView) view.findViewById(R.id.ieeemsh_id)).setText(IEEE_membership_ID+"");
+                ((TextView) view.findViewById(R.id.phone)).setText(phone_number);
+                ((TextView) view.findViewById(R.id.type)).setText(type);
+
+                ((TextView) view.findViewById(R.id.bdate)).setText(birthdate);
+                ((TextView) view.findViewById(R.id.gender)).setText(gender);
+            }
+
+            @Override
+            public void onFailure(int statusCode, JSONObject body) {
+
+                if(statusCode == 401) {
+                    MainActivity.logout();
+                }
+                else if(statusCode == 400)
+                {
+                    //don't remember which message

```

Comment:
in this controller if you got 400 that means that the application set the id wrong and in this case the user should get a server error message as it is not his fault